tests/test_coverage.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+"""Test the coverage and update the threshold when coverage is increased."""
+
+import os, re, shutil, subprocess
+import pytest
+
+def _get_current_coverage():
+    """Helper function that returns the coverage computed with kcov."""
+    kcov_ouput_dir = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "kcov_output"
+    )
+
+    # By default the build output for kcov and unit tests are both in the debug
+    # directory. This causes some linker errors that I haven't investigated.
+    # Error: error: linking with `cc` failed: exit code: 1
+    # An easy fix is to have separate build directories for kcov & unit tests.
+    kcov_build_dir = os.path.join(
+        os.path.dirname(os.path.realpath(__file__)),
+        "kcov_build"
+    )
+
+    # Remove kcov output and build directory to be sure we are always working
+    # on a clean environment.
+    shutil.rmtree(kcov_ouput_dir, ignore_errors=True)
+    shutil.rmtree(kcov_build_dir, ignore_errors=True)
+
+    exclude_pattern = (
+        '${CARGO_HOME:-$HOME/.cargo/},'
+        'usr/lib/,'
+        'lib/'
+    )
+    exclude_region = "'mod tests {'"
+
+    kcov_cmd = "CARGO_TARGET_DIR={} cargo kcov --all " \
```

Comment:
you need to get the coverage for all features. cargo kcov --all --features=elf,bzImage